### PR TITLE
[C++ for OpenCL] Added references section.

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
The document is updated to cite the references where needed.

### CODE REVIEW

File: cxx4opencl/address_spaces.txt [hunk 1]
```diff
@@ -10,8 +10,8 @@
 This section only documents behavior related to {cpp} features. For example
 conversion rules are extended from the qualification conversion {cpp}17
 [conv.qual] but the compatibility is determined using notation of sets and
-overlapping of address spaces from Embedded C
-(`ISO/IEC JTC1 SC22 WG14 N1021 s3.1.3`). For OpenCL it means that implicit
+overlapping of address spaces from `section 3.1.3` of <<embedded-c-spec,
```
Comment (mantognini) [inline cxx4opencl/address_spaces.txt:13]: In n1021 this is indeed section 3.1.3 but in the published version it's 5.1.3.

**Review by mantognini — CHANGES_REQUESTED** [review]
LGTM, a minor detail only.

File: cxx4opencl/address_spaces.txt (hunk 1 above)
Comment (AnastasiaStulova) [reply cxx4opencl/address_spaces.txt]: Good spot! I have updated it now. Thanks.

**Review by mantognini — APPROVED** [review]
LGTM

**Review by svenvh — APPROVED** [review]

**Pull request merged**